### PR TITLE
built_in string right

### PR DESCRIPTION
**Pull request by cuiyuan**
solve this issue,
https://github.com/pingcap/tidb/issues/3074

### CODE REVIEW

File: expression/builtin_string.go [hunk 1]
```diff
@@ -318,7 +319,42 @@ type builtinRightSig struct {
...
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	sc := new(variable.StatementContext)
+	sc.IgnoreTruncate = true
+	length, err := arg1.ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if length <= 0 {
```
Comment (woodpenker) [inline expression/builtin_string.go]: here is what I impl , small diff to you
```
l := int(length)
	if l < 0 {
		l = 0
	} else if l > len(str) {
		l = len(str)
	}
	d.SetString(str[len(str)-l:])
```

File: expression/builtin_string.go [hunk 2]
```diff
@@ -318,7 +319,42 @@ type builtinRightSig struct {
 }
 
 func (b *builtinRightSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("RIGHT")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
```
Comment (shenli) [inline expression/builtin_string.go]: return d, errors.Trace(err)

**Comment by shenli** [comment]
@cuiyuan Please update the `plan/typeinferer.go`.

**Comment by shenli** [comment]
LGTM
@cuiyuan Thanks for your PR!

**Pull request merged**